src/java.base/share/classes/java/lang/Process.java

```diff
@@ -612,6 +618,73 @@ public boolean waitFor(Duration duration) throws InterruptedException {
      */
     public abstract int exitValue();
 
+    /**
+     * Close all writer and reader streams and terminate the process.
+     * The streams are closed immediately and the process is terminated without waiting.
+     * <p>
+     * Before calling {@code close} the caller should read the streams for any
+     * data or text and call {@linkplain #waitFor() waitFor} if the exit value is needed.
+     * The contents of streams that have not been read fully are lost;
+     * they are discarded or ignored.
+     * Streams should be {@code closed} when no longer needed.
+     * Closing an already closed stream usually has no effect but is specific to the stream.
+     * IOExceptions that occur when closing streams are ignored.
+     * <p>
+     * The process may already have exited or be in the process of exiting;
+     * if it is {@linkplain #isAlive() alive}, it is {@linkplain #destroy destroyed}.
+     * IOExceptions that occur when destroying the process are ignored.
+     * <p>
+     * Example using try-with-resources writing text to a process, reading back the
+     * response, and closing the streams and process:
+     * {@snippet class=ProcessExamples region=example}
+     *
+     * @implSpec
+     * The outputWriter and outputStream to the process are closed.
+     * The inputReader and inputStream from the process are closed.
+     * The errorReader and errorStream from the process are closed.
+     * The process is destroyed.
+     * @since 26
+     */
+    public void close() {
+        List<Closeable> closeable = new ArrayList<>();
+        closeable.add(outputWriter != null ? outputWriter : getOutputStream());
+        closeable.add(inputReader != null ? inputReader : getInputStream());
+        closeable.add(errorReader != null ? errorReader : getErrorStream());
+
+        // close each and capture any exceptions
+        List<IOException> exceptions = closeable.stream()
+                .map(Process::doClose)
+                .filter((Objects::nonNull))
+                .toList();
+
+        if (!exceptions.isEmpty()) {
+            // TBD: Log exceptions closing streams
+            IOException ex = new IOException("exception closing process streams");
+            exceptions.forEach((e) -> ex.addSuppressed(e));
+        }
+
+        // Wait briefly for process to exit, if not exited immediately, destroy
+        try {
+            boolean alive = waitFor(Duration.ofMillis(2000));
```

Comment:
Doesn't this go against the `the process is terminated without waiting` statement in the spec?